src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,32 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Reverts the {@code model}'s address book to its previous state.
+ * @author {damithc}-reused
+ *     adapted from <a href="https://github.com/se-edu/addressbook-level4">ab4</a>.
+ */
+public class UndoCommand extends Command {
+
+    public static final String COMMAND_WORD = "undo";
+    public static final String MESSAGE_SUCCESS = "Last command have been undone!";
+    public static final String MESSAGE_FAILURE = "No commands to undo!";
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) throws CommandException {
+        requireNonNull(model);
+
+        if (!model.hasHistory()) {
+            throw new CommandException(MESSAGE_FAILURE);
+        }
+
+        model.undoAddressBook();
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
```

Comment:
Since we do not have a command to view all persons even, it seems a bit weird to show all persons after an undo. It might be better to revert the filtered list to whatever list (or even personType) it was showing during the state we are undo-ing to if possible. 

At the very least we can also default to showing all patients like it does on start up, as from what I can see the states are being stored without filtered list information.

Reply:
Makes sense, I have removed the line entirely in my recent commit